packages/react-chat/src/views/ChatWindow/index.tsx

```diff
@@ -42,7 +48,12 @@ const ChatWindow: React.FC = () => {
         startTime={state.session.startTime}
         hasEnded={runtime.isStatus(SessionStatus.ENDED)}
         isLoading={!state.session.turns.length}
-        onStart={runtime.launch}
+        onStart={async () => {
+          if (!session.autostart) {
+            setAutostart(true);
+          }
+          await runtime.launch();
+        }}
```

Comment:
since all of this state is already in `runtime` we can probably move all of this logic into the `runtime.launch` and keep this as just `onStart={runtime.launch}`

Reply:
refactored this since I believe autostart shouldn't belong to session, it's a general chat config